node/src/accountant/payment_adjuster/service_fee_adjuster.rs

```diff
@@ -302,38 +279,76 @@ mod tests {
                 weight_1,
-                multiple_by_billion(1_800_000_000),
+                multiply_by_quintillion_concise(1.8),
             ),
             AdjustedAccountBeforeFinalization::new(
                 account_2
-                    .weighted_account
+                    .weighed_account
                     .analyzed_account
                     .qualified_as
                     .bare_account,
                 weight_2,
-                multiple_by_billion(4_200_000_000) - 1,
+                multiply_by_quintillion_concise(4.2) - 1,
             ),
             AdjustedAccountBeforeFinalization::new(
                 account_4
-                    .weighted_account
+                    .weighed_account
                     .analyzed_account
                     .qualified_as
                     .bare_account,
                 weight_4,
-                multiple_by_billion(500_000_000),
+                multiply_by_quintillion_concise(0.5),
             ),
         ];
         assert_eq!(thriving_competitors, expected_adjusted_outweighed_accounts)
     }
 }
+
+#[cfg(test)]
+pub mod illustrative_util {
+    use crate::accountant::payment_adjuster::miscellaneous::data_structures::WeighedPayable;
+    use crate::accountant::payment_adjuster::service_fee_adjuster::compute_unconfirmed_adjustments;
+    use thousands::Separable;
+    use web3::types::Address;
+
+    pub fn illustrate_why_we_need_to_prevent_exceeding_the_original_value(
+        cw_service_fee_balance_minor: u128,
+        weighed_accounts: Vec<WeighedPayable>,
+        wallet_of_expected_outweighed: Address,
+        original_balance_of_outweighed_account: u128,
+    ) {
+        let unconfirmed_adjustments =
+            compute_unconfirmed_adjustments(weighed_accounts, cw_service_fee_balance_minor);
+        // The results are sorted from the biggest weights down
+        assert_eq!(
+            unconfirmed_adjustments[1].wallet(),
+            wallet_of_expected_outweighed
+        );
+        // To prevent unjust reallocation we used to secure a rule an account could never demand
+        // more than 100% of its size.
+
+        // Later it was changed to a different policy, the so called "outweighed" account is given
+        // automatically a balance equal to its disqualification limit. Still, later on, it's quite
+        // likely to acquire slightly more by a distribution of the last bits of funds away from
+        // within the consuming wallet.
+        let proposed_adjusted_balance = unconfirmed_adjustments[1].proposed_adjusted_balance_minor;
+        assert!(
+            proposed_adjusted_balance > (original_balance_of_outweighed_account * 11 / 10),
+            "we expected the proposed balance at least 1.1 times bigger than the original balance \
+            which is {} but it was {}",
+            original_balance_of_outweighed_account.separate_with_commas(),
+            proposed_adjusted_balance.separate_with_commas()
+        );
+    }
```

Comment:
Intended for a different file (as you seem fit):

Also, it would be better if you can add the cases that might end-up in the real-world, using the diagrams or without it. It's up to you wherever you'd like to add.

You might like to to explain the recursion function with the role of the disq limit, and how the wallets balance is exhausted in the ending phase.